src/test/java/com/serotonin/bacnet4j/obj/ObjectTestUtils.java

```diff
@@ -0,0 +1,147 @@
+package com.serotonin.bacnet4j.obj;
+
+import static org.junit.Assert.fail;
+
+import java.time.Clock;
+import java.util.LinkedList;
+import java.util.List;
+
+import com.serotonin.bacnet4j.type.Encodable;
+import com.serotonin.bacnet4j.type.enumerated.PropertyIdentifier;
+
+public class ObjectTestUtils {
+    /**
+     * Test utility to wait for a specific value to be written to an object.
+     */
+    public static class ObjectWriteNotifier<T extends BACnetObject> extends AbstractMixin {
+        private final T bo;
+        private final boolean debug;
+        private final List<WrittenProperty> writtenProperties = new LinkedList<>();
```

Comment:
Why use a LinkedList here? `ArrayList` is better almost always (less memory, better performance). Since this is a test though it's definitely a nitpick, I'm more curious why you chose `LinkedList`. I suspect that it is because you are removing the first element, in which case you could actually use the `pop`/`removeFirst` method of `Deque` (`LinkedList` implements this).

Reply:
Yes, it's because it acts like a queue, and so it's faster when removing the first element. I'll change the method i use to do so.